backend-api/tests/api-tests/activeSession.test.ts

```diff
@@ -0,0 +1,119 @@
+import { randomUUID } from "crypto";
+import {
+  PROXY_API_INSTANCE,
+  SESSIONS_API_INSTANCE,
+  STS_MOCK_API_INSTANCE,
+} from "./utils/apiInstance";
+import { getFirstRegisteredClient } from "./utils/getRegisteredClient";
+
+describe("GET /async/activeSession", () => {
+  describe("Given service token is missing in the request header", () => {
+    it("Returns an error and 401 status code", async () => {
+      const response = await SESSIONS_API_INSTANCE.get("/async/activeSession");
+
+      expect(response.status).toBe(401);
+      expect(response.data).toStrictEqual({
+        error: "unauthorized",
+        error_description: "Invalid authorization header",
+      });
+    });
+  });
+
+  describe("Given service token is invalid", () => {
+    it("Returns an error and 400 status code", async () => {
+      const response = await SESSIONS_API_INSTANCE.get("/async/activeSession", {
+        headers: { Authorization: "Bearer one.two.three.four.five" },
+      });
+
+      expect(response.status).toBe(400);
+      expect(response.data).toStrictEqual({
+        error: "invalid_request",
+        error_description: "Failed to decrypt service token",
+      });
+    });
+  });
+
+  describe("Given active session is not found", () => {
+    it("Returns an error and 404 status code", async () => {
+      const accessToken = await getAccessToken(
+        "478E3BF8-C3D5-4EA2-A6EE-38B07F2EE0FD",
+      );
+
+      const response = await SESSIONS_API_INSTANCE.get("/async/activeSession", {
+        headers: { Authorization: `Bearer ${accessToken}` },
+      });
+
+      expect(response.status).toBe(404);
+      expect(response.data).toStrictEqual({
+        error: "session_not_found",
+        error_description:
+          "No active session found for the given sub identifier",
+      });
+    });
+  });
+
+  describe("Given a successful call is made to /async/activeSession", () => {
```

Comment:
Can this maybe be something like "Given the request is valid and a session is found?" or just "Given a session is found"?

Reply:
Or even "Given there's an active session for this user" 👀

Reply:
Yeah, nice - updated